### PR TITLE
[MatrixCamp] add function sin

### PR DESCRIPTION
**Pull request by hiyoyolumi**
**What type of PR is this?**

- [ ] API-change
- [ ] BUG
- [ ] Improvement
- [ ] Documentation
- [x] Feature
- [ ] Test and CI
- [ ] Code Refactoring

**Which issue(s) this PR fixes:**

issue #1966

**What this PR does / why we need it:**

implement Ceil builtin function

**Special notes for your reviewer:**

Not Available

**Additional documentation (e.g. design docs, usage docs, etc.):**

Not Available

### CODE REVIEW

**Comment by m-schen** [comment]
need license related information at the header of code files.  @hiyoyolumi

File: pkg/builtin/unary/sin.go [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+				return resultVector, nil
+			},
+		},
+		{
+			Typ: types.T_float64,
+			ReturnType: types.T_float64,
+			Fn: func(origVec *vector.Vector, proc *process.Process, _ bool) (*vector.Vector, error) {
+				origVecCol := origVec.Col.([]float64)
+				resultVector, err := process.Get(proc, 8*int64(len(origVecCol)), types.Type{Oid: types.T_float64, Size: 8})
+				if err != nil {
```
Comment (broccoliSpicy) [inline pkg/builtin/unary/sin.go:84]: perhaps we could reuse the original vector when possible

**Review by broccoliSpicy — CHANGES_REQUESTED** [review]

**Comment by hiyoyolumi** [comment]
Ok, I will improve it later.





------------------ Original ------------------
From: broccoliSpicy ***@***.***&gt;
Date: Mon,Mar 28,2022 2:01 PM
To: matrixorigin/matrixone ***@***.***&gt;
Cc: hiyoyolumi ***@***.***&gt;, Mention ***@***.***&gt;
Subject: Re: [matrixorigin/matrixone] [MatrixCamp] add function sin() (PR#2020)





 
@broccoliSpicy requested changes on this pull request.
 
 
In pkg/builtin/unary/sin.go:
 &gt; +} +results := encoding.DecodeFloat64Slice(resultVector.Data) +results = results[:len(origVecCol)] +resultVector.Col = results +nulls.Set(resultVector.Nsp, origVec.Nsp) +vector.SetCol(resultVector, sin.SinFloat32(origVecCol, results)) +return resultVector, nil +}, +}, +{ +Typ: types.T_float64, +ReturnType: types.T_float64, +Fn: func(origVec *vector.Vector, proc *process.Process, _ bool) (*vector.Vector, error) { +origVecCol := origVec.Col.([]float64) +resultVector, err := process.Get(proc, 8*int64(len(origVecCol)), types.Type{Oid: types.T_float64, Size: 8}) +if err != nil {  
perhaps we could reuse the original vector when possible

File: pkg/builtin/unary/sin.go (hunk 1 above)
Comment (hiyoyolumi) [reply pkg/builtin/unary/sin.go:89]: I have done this.

**Review by domingozhang — COMMENTED** [review]
bvt passed

**Pull request merged**